packages/workflow/test/Expression.test.ts

```diff
@@ -173,7 +173,18 @@ describe('Expression', () => {
 		const expression = new Expression(workflow);
 
 		const evaluate = (value: string, data: INodeExecutionData[]) => {
-			return expression.getParameterValue(value, null, 0, 0, 'node', data, 'manual', '', {});
+			const itemIndex = data.length === 0 ? -1 : 0;
+			return expression.getParameterValue(
+				value,
+				null,
+				0,
+				itemIndex,
+				'node',
+				data,
+				'manual',
+				'',
+				{},
+			);
```

Comment:
After the change, [this test](https://github.com/n8n-io/n8n/blob/e8b51c8da9beecc1f4f95eb9528c8aa29f0f7222/packages/workflow/test/ExpressionFixtures/base.ts#L265-L269)

```ts
{
	type: 'evaluation',
	input: [],
	output: undefined,
},
```
started being caught by this check

```ts
if (executionData.length <= that.itemIndex) {
	throw new ExpressionError(`No data found for item-index: "${that.itemIndex}"`, {
		runIndex: that.runIndex,
		itemIndex: that.itemIndex,
	});
}
```

Because the itemIndex was hardcoded to `0`.